datasetinsights/estimators/faster_rcnn.py

```diff
@@ -22,6 +22,605 @@
+        dataset: dataset obj must have len and __get_item__
+        is_train: whether or not the sampler is for training data
+
+    Returns:
+        data_sampler: (torch.utils.data.Sampler)
+
+    """
+    if is_distributed:
+        sampler = torch.utils.data.distributed.DistributedSampler(
+            dataset, shuffle=is_train
+        )
+
+    else:
+        if is_train:
+            sampler = torch.utils.data.RandomSampler(dataset)
+        else:
+            sampler = torch.utils.data.SequentialSampler(dataset)
+    return sampler
+
+
+def load_data(
+    config,
+    dataset,
+    sampler,
+    train,
+    *,
+    batch_size=1,
+    num_workers=0,
+    collate_fn=None,
+):
+    """
+    load dataset and create dataloader
+    Args:
+        config: (CfgNode): estimator config:
+        dataset: dataset obj must have len and __get_item__
+        sampler: (torch.utils.data.Sampler)
+        train: whether or not the sampler is for training data
+        batch_size: batch_size
+        num_workers: num_workers
+        collate_fn: Prepare batch to be format Faster RCNN expects
+    Returns data_loader: torch.utils.data.DataLoader
+
+    """
+    if config.system.distributed:
+        if train:
+            batch_sampler = torch.utils.data.BatchSampler(
+                sampler, batch_size, drop_last=False
+            )
+
+            loader = torch.utils.data.DataLoader(
+                dataset,
+                batch_sampler=batch_sampler,
+                num_workers=num_workers,
+                collate_fn=collate_fn,
+            )
+            return loader
+        else:
+            loader = torch.utils.data.DataLoader(
+                dataset,
+                batch_size=batch_size,
+                sampler=sampler,
+                num_workers=num_workers,
+                collate_fn=collate_fn,
+                drop_last=False,
+            )
+            return loader
+    else:
+        logger.info(f"not creating distributed dataloader")
+        data_loader = torch.utils.data.DataLoader(
+            dataset,
+            batch_size=batch_size,
+            sampler=sampler,
+            num_workers=num_workers,
+            drop_last=train,
+            collate_fn=collate_fn,
+        )
+        return data_loader
+
+
+def create_dataloader(config, dataset, sampler, split):
+    """
+    initiate data loading
+    Args:
+        config: (CfgNode): estimator config:
+        dataset: dataset obj must have len and __get_item__
+        sampler: (torch.utils.data.Sampler)
+        split: train, val, test
+    Returns data_loader: torch.utils.data.DataLoader
+
+    """
+    is_train = False
+    if split == TRAIN:
+        is_train = True
+    dataloader = load_data(
+        config=config,
+        dataset=dataset,
+        batch_size=config[split].batch_size,
+        sampler=sampler,
+        collate_fn=collate_fn,
+        train=is_train,
+    )
+    return dataloader
+
+
+def create_optimizer_lrs(config, model):
```

Comment:
Should this method be a module level method? Should it be static method for the FasterRCNN class?

Reply:
Fixed